### DISCUSSION TITLE
Presentation of profile view on front end

### DISCUSSION THREAD

**Discussion by code-blooded-guy** [Sunbird-RC/community#309] [Sunbird RC Technical Specification Feedback]
I need to change the presentation of the profile page.
- Need proper field positioning to show on the block

Screenshot attached
![screenshot_profile](https://user-images.githubusercontent.com/59663471/207847854-aea40e40-7ea3-489b-8269-1a965d2c50e2.png)

**Comment by parthlawate** [comment]
So the need from backend is to have ability to order schema fields ?

**Comment by code-blooded-guy** [comment]
Yes, if possible from the backend or someone has already done this on the front end via layout config also works.

**Reply by parthlawate** [reply]
I think having an ability to order fields in the schema in the backend & the frontend dynamically rendering it would be better and more automatic . cc  @tejash-jl @Renuka-S

**Reply by tejash-jl** [reply]
@parthlawate @code-blooded-guy 
Ordering of fields in the schema is more of a front-end config. From a backend perspective, the order really doesn't matter.

**Reply by parthlawate** [reply]
Since ideally We'd want the frontend to need minimum config. Pick up max stuff from backend , would be good to have this ability. However i see where you are coming from.. Maybe a light middleware layer could be done for this ? 

Even in admin portal stories, a user might want to reorder fields in the schema and expect them to reflect in the form for instance

**Reply by parthlawate** [reply]
How about adding a system field uiconfig (type object) where the ordering and other details like rows/columns can be maintained. This can be used by the stock RC UI framework to affect how the field is shown in the UI.

If the implementer is using their own UI they may put their own properties in this object.

This will let adopters maintain the schema and config in a single place. 

@tejash-jl

**Reply by pramodkvarma** [reply]
Do keep the UX config seperated (and optional) from core registry config. UX config can be a layer on top and can refer to registry fields.

**Reply by parthlawate** [reply]
So the UX config should be part of the reference Sunbird UI to be used as a best practice. 

Generally wherever the registry has a UI - there will be forms as well as record display pages. We want to avoid hard coding any of this as static forms/ pages.. the idea of this kind of UX/UI config is to be able to get to a point where forms & pages are autogenerated.

**Reply by pramodkvarma** [reply]
Makes sense.
